lib/global_uid/allocator.rb

```diff
@@ -0,0 +1,66 @@
+module GlobalUid
+  class Allocator
+    attr_reader :allocations, :max_window_size, :incrementing_by, :connection
+
+    def initialize(max_window_size: 5, incrementing_by:, connection:)
+      @allocations = []
+      @max_window_size = max_window_size
+      @incrementing_by = incrementing_by
+      @connection = connection
+      validate_connection_increment
+    end
+
+    def allocate_one(table)
+      identifier = connection.insert("REPLACE INTO #{table} (stub) VALUES ('a')")
+      allocate(identifier)
+    end
+
+    def allocate_many(table, count:)
+      increment_by = validate_connection_increment
+
+      start_id = connection.insert("REPLACE INTO #{table} (stub) VALUES " + (["('a')"] * count).join(','))
+      identifiers = start_id.step(start_id + (count - 1) * increment_by, increment_by).to_a
+      identifiers.each { |identifier| allocate(identifier) }
+      identifiers
+    end
+
+    private
+
+    def allocate(identifier)
+      allocations.shift if allocations.size >= max_window_size
+      allocations << identifier
+
+      if !valid_allocation?
+        db_increment = connection.select_value("SELECT @@auto_increment_increment")
+        message = "Configured: '#{incrementing_by}', Found: '#{db_increment}' on '#{connection.current_database}'. Recently allocated IDs: #{allocations}"
+        alert(InvalidIncrementException.new(message))
+      end
```

Comment:
I was hoping we'd be checking the other config as well. This does not prevent - for example, two different alloc id servers, with the the right increment_increment, but same (bad) increment_offset. Pod15 incident had both of these misconfigured.

Reply:
The `increment_offset` was never a configuration option and is unique across the allocation servers. The code in it's current state would have required a lot more changes to receive and validate that option which is why I held off. 
I agree with you though, it should be something that this gem receives and validates, without it, the concept falls over. It's something I'd like to see this gem do too so I'll add it to #70